src/galaxy/StarSystemGenerator.cpp

```diff
@@ -1290,13 +1290,9 @@ SystemBody* StarSystemRandomGenerator::MakeGravPointForBodies(RefCountedPtr<Star
 	return grav;
 }
 
-std::string StarSystemRandomGenerator::GetStarNameFromId(const int id)
+char StarSystemRandomGenerator::GetStarSuffixFromOffset(const int offsetId) const
```

Comment:
Maybe this would help?
```C++
std::string StarSystemRandomGenerator::GetStarSuffixFromOffset(const int offsetId) const
{
    return std::string(1, 'A' + Clamp(offsetId, 0, 26));
}
```

Reply:
I'd like to avoid constructing a `std::string` then return it to a method taking a reference when all you want from this is the single character whcih you _might_ then use in a `std::string`

Reply:
> I'd like to avoid constructing a `std::string`

I see from patch you're constructing it anyway.
Better do it inside `GetStarSuffixFromOffset` to prevent typos since common code is used *twice*

When having 26+ stars, we'll just have to rewrite this to generate (...X, Y, Z, AA, AB, AC...)
And rewrite GravPoint names to make them joinable via comma instead of concatenating (not sure which is the case now)

**UPD**
suffixes are joined via comma, allowing AA, AB, AC... while E:D just concatenates them (but I've never seen more than 5 stars in that game - 11+ brown dwarfs around main sequence star counts as planets)

Reply:
No I mean that I would like to avoid constructing the `std::string` as a return value, it's fine if you construct one outside but this is really a wrapper around `'A' + int` which gives you a single char that you don't have to use with a `std::string` and you could refactor to work with insert/replace on a string.

Refactoring is for a future that never comes, also it should only be used once.

Reply:
I think once you're talking more than 5 or 6 stars in a single system they have to be so distant from one another that defining them as a single star system gets challenging to the meaning of the word.

They'd have to be light years apart and in reality would be affected by other star systems as much as by the other stars they're supposed to be in a system with.

Reply:
The record for E:D is: 11 stars
https://spansh.co.uk/system/5645877119803#system-bodies
https://spansh.co.uk/system/527009230954354#system-bodies
https://spansh.co.uk/system/40122780307#system-bodies

Reply:
That'd only get us to `K`, I think we're safe with 14 remaining letters :D